### DISCUSSION TITLE
Allow to pass a custom Id

### DISCUSSION THREAD

**Discussion by josemrodrigu** [gpbl/react-day-picker#1551] [Ideas]
We are in the need of use our own `id` instead the one generated by the [genId function](https://github.com/gpbl/react-day-picker/blob/1441c6638f6e12995776c2bdd0b51c7d91d23f06/packages/react-day-picker/src/hooks/useId/useId.ts#L109).

I can see that [useId](https://github.com/gpbl/react-day-picker/blob/1441c6638f6e12995776c2bdd0b51c7d91d23f06/packages/react-day-picker/src/hooks/useId/useId.ts#L109) accepts a default `id`provided in props, but actually it's never used.

I can see that the only time this hook is executed is [here](https://github.com/gpbl/react-day-picker/blob/a8c3e3c1565ff2fc729825b2f337a6323bbaae2b/packages/react-day-picker/src/components/Month/Month.tsx#L20), at the time that `captionId` is generated. Should it be possible to pass an id received in props, isn't it?
I think we can store this `id` (optional, of course) directly into [DayPickerContext](https://github.com/gpbl/react-day-picker/blob/a8c3e3c1565ff2fc729825b2f337a6323bbaae2b/packages/react-day-picker/src/contexts/DayPicker/DayPickerContext.tsx).

Let me know about your thoughts, I'll be glad to contribute.

Thanks!

**Comment by gpbl** [comment]
Implementing this in https://github.com/gpbl/react-day-picker/pull/1556
